src/engraving/dom/instrtemplate.cpp

```diff
@@ -853,19 +853,28 @@ const InstrumentTemplate* searchTemplateForInstrNameList(const std::list<String>
 {
     const InstrumentTemplate* bestMatch = nullptr; // default if no matches
     int bestMatchStrength = 0; // higher for better matches
-    for (const InstrumentGroup* g : instrumentGroups) {
-        for (const InstrumentTemplate* it : g->instrumentTemplates) {
-            for (const String& name : nameList) {
-                if (name.isEmpty() || it->useDrumset != useDrumset) {
+
+    for (String name : nameList) {
+        if (name.isEmpty()) {
+            continue;
+        }
+
+        if (!caseSensitive) {
+            name = name.toLower();
+        }
+        StaffName instrName(name);
+
+        for (const InstrumentGroup* g : instrumentGroups) {
+            for (const InstrumentTemplate* it : g->instrumentTemplates) {
+                if (it->useDrumset != useDrumset) {
                     continue;
                 }
+
                 String trackName = it->trackName;
```

Comment:
Avoiding another string copy:
```C++
const String& trackName = caseSensitive ? it->trackName : it->trackName.toLower();
```

Reply:
`String trackName = it->trackName;`
muse::String is designed like QString from QT. It has move semantic. Copy of muse::String object doesn't lead to allocate memory. Only just pointer to shared memory is copied. Allocating memory is happening only if muse::String data is changed.


`const String& trackName = it->trackName.toLower();` that code doesn't improve performance. memory is allocated inside of `toLower()` method.  But `toLower()` creates local variable and return it. So we get reference to temporary local object here which is potentially is not safe. Although modern compiler like gcc or clang can handle it and we don't get crash here. But I am not sure about more young compiler like Emscrypten for example. Anyway as I remember address sanitizer marks that code as a mistake.

Reply:
But I cant change it if you insist.

Reply:
No, I'm happy with it. Thanks for the explanation!